packages/ui/src/modals/TransferModal/SignTransferModal.tsx

```diff
@@ -67,14 +60,20 @@ export function SignTransferModal({ onClose, from, amount, to }: Props) {
       onClose()
     }
 
-    if (keyringPair.meta.isInjected) {
-      const { signer } = await web3FromAddress(from.address)
+    let subscription: Subscription
 
-      await submittableExtrinsic.signAndSend(from.address, { signer: signer }, statusCallback).catch(console.error)
+    if (keyringPair.meta.isInjected) {
+      web3FromAddress(from.address).then(({ signer }) => {
```

Comment:
I guess it's still ok to use `await` here.

Reply:
I'm not sure if I can return Promise from the `useEffect()` hook callback (using the `async` keyword). I wanted to return the `subscription` callback so I've used the synchronous arrow funciton.